include-refactoring/xsimd/config/xsimd_cpuid.hpp

```diff
@@ -1,139 +1,151 @@
 #ifndef XSIMD_CPUID_HPP
 #define XSIMD_CPUID_HPP
 
-#include "../types/xsimd_all_registers.hpp"
 #include <cstring>
 #include <algorithm>
 #include <array>
 
+#include "../types/xsimd_all_registers.hpp"
 
-namespace xsimd {
-  namespace detail {
-
-    struct supported_arch
+namespace xsimd
+{
+    namespace detail
     {
-        unsigned sse : 1;
-        unsigned sse2 : 1;
-        unsigned sse3 : 1;
-        unsigned ssse3 : 1;
-        unsigned sse4_1 : 1;
-        unsigned sse4_2 : 1;
-        unsigned sse4a : 1;
-        unsigned fma3 : 1;
-        unsigned fma4 : 1;
-        unsigned xop : 1;
-        unsigned avx : 1;
-        unsigned avx2 : 1;
-        unsigned avx512 : 1;
-        unsigned neon : 1;
-        unsigned neon64 : 1;
-
-        // version number of the best arch available
-        unsigned best;
-
-        supported_arch()
+        struct supported_arch
         {
-             memset(this, 0, sizeof(supported_arch));
+            unsigned sse : 1;
+            unsigned sse2 : 1;
+            unsigned sse3 : 1;
+            unsigned ssse3 : 1;
+            unsigned sse4_1 : 1;
+            unsigned sse4_2 : 1;
+            unsigned sse4a : 1;
+            unsigned fma3 : 1;
+            unsigned fma4 : 1;
+            unsigned xop : 1;
+            unsigned avx : 1;
+            unsigned avx2 : 1;
+            unsigned avx512 : 1;
+            unsigned arm7 : 1;
+            unsigned arm8_32 : 1;
+            unsigned arm8_64 : 1;
+
+            // version number of the best arch available
+            unsigned best;
+
+            supported_arch()
+            {
+                 memset(this, 0, sizeof(supported_arch));
 
 #if defined(__aarch64__) || defined(_M_ARM64)
-             // neon is required on AArch64
-             neon = 1;
-             neon64 = 1;
-             best = neon64::version();
-
-#elif defined(__ARM_NEON) || defined(_M_ARM)
-             neon = bool(getauxval(AT_HWCAP) & HWCAP_NEON);
-             best = neon::version() * neon;
+                 arm7 = 1;
+                 arm8_32 = 1;
+                 arm8_64 = 1;
+                 best = arm8_64::version();
+#elif defined(__ARM_NEON)
+#if __ARM_ARCH >= 8
+                 arm7 = 1;
+                 arm8_32 = 1;
+                 arm8_64 = 0;
+                 best = arm8_32::version();
+#else
+                 arm7 = 1;
+                 arm8_32 = 0;
+                 arm8_64 = 0;
+                 best = arm7::version();
+#endif
+//#elif defined(__ARM_NEON) || defined(_M_ARM)
+//                 neon = bool(getauxval(AT_HWCAP) & HWCAP_NEON);
```

Comment:
why did you remove the runtime detection code?

Reply:
It's just for testing for now, I'm not sure how we should fill the different fields. There were only neon and neon64 while we have arm7, arm8_32 and arm8_64 registers. But arm8_32 is supposed to be fully compatible with arm7, so I'm don't know if we should keep both of simplify (although I find keeping both more explicit).